File: JDK11/opentraceability/src/main/java/opentraceability/models/events/EventILMD.java
package opentraceability.models.events;

import opentraceability.interfaces.IEventKDE;
import opentraceability.Constants;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityExtensionElementsAttribute;
import opentraceability.utility.attributes.OpenTraceabilityArrayAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;
import opentraceability.utility.attributes.OpenTraceabilityObjectAttribute;
import opentraceability.utility.attributes.OpenTraceabilityRepeatingAttribute;
import opentraceability.utility.Country;

import java.time.OffsetDateTime;
import java.util.List;

import opentraceability.models.events.kdes.CertificationList;

public class EventILMD {
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name="productionMethodForFishAndSeafoodCode")
    @OpenTraceabilityJsonAttribute(name="cbvmda:productionMethodForFishAndSeafoodCode")
    public String productionMethodForFishAndSeafoodCode;

    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name="itemExpirationDate")
    @OpenTraceabilityJsonAttribute(name="cbvmda:itemExpirationDate")
    public OffsetDateTime itemExpirationDate;

    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name="productionDate")
    @OpenTraceabilityJsonAttribute(name="cbvmda:productionDate")
    public OffsetDateTime productionDate;

    @OpenTraceabilityArrayAttribute(itemType = Country.class)
    @OpenTraceabilityRepeatingAttribute
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name="countryOfOrigin")
    @OpenTraceabilityJsonAttribute(name="cbvmda:countryOfOrigin")
    public List<Country> countryOfOrigin;

    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name="lotNumber")
    @OpenTraceabilityJsonAttribute(name="cbvmda:lotNumber")
    public String lotNumber;

    @OpenTraceabilityObjectAttribute
    @OpenTraceabilityAttribute(ns = Constants.CBVMDA_NAMESPACE, name="certificationList")
    @OpenTraceabilityJsonAttribute(name="cbvmda:certificationList")
    public CertificationList certificationList;

    @OpenTraceabilityExtensionElementsAttribute
    public List<IEventKDE> extensionKDEs;
}

File: Java/OpenTraceability/src/main/opentraceability/models/events/eventilmd.kt
package models.events
import interfaces.IEventKDE
import java.util.*
import models.events.kdes.CertificationList
import utility.Country
import java.time.OffsetDateTime
class EventILMD {

    //TODO: review this

    //[OpenTraceability(Constants.CBVMDA_NAMESPACE, "productionMethodForFishAndSeafoodCode")]
    //[OpenTraceabilityJson("cbvmda:productionMethodForFishAndSeafoodCode")]
    var ProductionMethodForFishAndSeafoodCode: String? = null

    //[OpenTraceability(Constants.CBVMDA_NAMESPACE, "itemExpirationDate")]
    //[OpenTraceabilityJson("cbvmda:itemExpirationDate")]
    var ItemExpirationDate: OffsetDateTime? = null


    //[OpenTraceability(Constants.CBVMDA_NAMESPACE, "productionDate")]
    //[OpenTraceabilityJson("cbvmda:productionDate")]
    var ProductionDate: OffsetDateTime? = null

    //[OpenTraceabilityArray]
    //[OpenTraceabilityRepeatingAttribute]
    //[OpenTraceability(Constants.CBVMDA_NAMESPACE, "countryOfOrigin")]
    //[OpenTraceabilityJson("cbvmda:countryOfOrigin")]
    var CountryOfOrigin: ArrayList<Country> = ArrayList<Country>()

    //[OpenTraceability(Constants.CBVMDA_NAMESPACE, "lotNumber")]
    //[OpenTraceabilityJson("cbvmda:lotNumber")]
    var LotNumber: String? = null

    //[OpenTraceabilityObject]
    //[OpenTraceability(Constants.CBVMDA_NAMESPACE, "certificationList")]
    //[OpenTraceabilityJson("cbvmda:certificationList")]
    var CertificationList: CertificationList? = null

    //[OpenTraceabilityExtensionElements]
    var ExtensionKDEs: ArrayList<IEventKDE> = ArrayList<IEventKDE>()
}
